columnflow/columnar_util.py

```diff
@@ -1599,6 +1614,35 @@ def _get_produced_columns(self, call_cache: set | None = None) -> set[str]:
     def produced_columns(self) -> set[str]:
         return self._get_produced_columns()
 
+    def _check_columns(self, ak_array: ak.Array, io_flag: IOFlag) -> None:
+        """
+        Check if awkward array contains at least one column matching each
+        entry in 'uses' or 'produces' and raise Exception if none were found.
+        """
+
+        columns = self._get_columns(
+            io_flag=io_flag,
+            # only check own columns
+            no_dependencies=True,
```

Comment:
Would be `dependencies=False` with the adjustment above.

Reply:
OK.